src/common/ufo-scarray.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2017 Karlsruhe Institute of Technology
+ *
+ * This file is part of Ufo.
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ufo-scarray.h"
+
+struct _UfoScarray {
+    GValueArray *array;
+};
+
+UfoScarray *
+ufo_scarray_new (guint n_dims, GType type)
```

Comment:
I like this commit a lot. Tiny thing, the first parameter is actually `num_elements` not the number of dimensions.